src/ModuleLoader.ts

```diff
@@ -715,7 +718,7 @@ function isNotAbsoluteExternal(
 	);
 }
 
-async function waitForDependencyResolution(loadPromise: LoadModulePromise) {
+async function waitForDependencyResolution(loadPromise: LoadModulePromise): Promise<void> {
 	const [resolveStaticDependencyPromises, resolveDynamicImportPromises] = await loadPromise;
-	return Promise.all([...resolveStaticDependencyPromises, ...resolveDynamicImportPromises]);
+	await Promise.all([...resolveStaticDependencyPromises, ...resolveDynamicImportPromises]);
```

Comment:
To avoid the unnecessary "await as last statement", one could also just type this as `Promise<unknown>`. My understanding is that this avoids one implicit layer of Promise unwrapping the compiler has to do.

Reply:
probably so (albeit probably microscopic in the grand schema of things). I believe I saw this pattern across the code base. I guess there is no "good" way to return `undefined` if the caller ignores the function result entirely. I think it makes the code easier to read and also interpret the intend. if the only caller of a function which returns a result ignores the result it feels more like an oversight than done on purpose. that's all.

I'll revert.